src/processing/mangekanter/mangekanter.md

```diff
@@ -0,0 +1,468 @@
+
+    ```processing
+    void draw() {
+      x += xFart;
+      y += yFart;
+      
+      if (x < 0) {
+        xFart = -xFart;
+      }
+      
+      if (x > width - 100) {
+        xFart = -xFart;
+      }
+      
+      if (y < 0) {
+        yFart = -yFart;
+      }
+      
+      if (y > height - 100) {
+        yFart = -yFart;
+      }
+    
+      background(0);
+      triangle(x, y, x + 100, y, x + 50, y + 100);
+    }
+    ```
+    
+    Her har vi tatt i bruk `triangle` istedenfor `rect`. Denne tar
+    imot seks argumenter, to for hvert hjørne i trekanten. `x, y` er
+    posisjonen til det første hjørnet øverst til venstre, `x + 100, y`
+    er posisjonen til det øverste høyre hjørnet og `x + 50, y + 100`
+    er det siste hjørnet nederst i midten.
+
++ Lagre programmet som *Trekant* ved å velge **File -> Save as** eller
+  trykke **Shift + Ctrl + S**.
+
++ Kjør programmet.
+
+![](SprettendeTrekant1.gif "En hvit trekant som peker nedover, spretter rundt på en svart bakgrunn.")
+
+## Forbedre leseligheten {.protip}
+
+Noen ganger kan det være vanskelig å lese kall på funksjoner som tar
+mange argumenter. I Processing tar de fleste funksjoner bare imot noen
+få argumenter, men `triangle` og et par andre tar seks eller flere. Da
+kan det være nyttig å dele opp kallet over flere linjer. For eksempel
+kunne setningen ovenfor vært skrevet slik at hvert hjørne var på hver
+sin linje:
+
+```processing
+triangle(x, y,
+  x + 100, y,
+  x + 50, y + 100);
+```
+
+Hvis man fortsatt synes det er vanskelig å lese eller rotete, kan man
+legge til noen ekstra mellomrom for å få ting på linje. Merk at om man
+bruker automatisk formatering av koden i Processing, vil den fjerne
+mellomrom den mener er overflødig.
+
+## Utfordringer {.try}
+
++ Kan du tegne trekanten motsatt vei, sånn at den ser ut som en
+  pil som peker oppover istedenfor nedover?
+
+![](SprettendeTrekant2.gif "En hvit trekant som peker oppover, spretter rundt på en svart bakgrunn.")
+
++ Kan du tegne to trekanter istedenfor en og lage en sekskantet
+  stjerne?
+
+![](SprettendeSekskantetStjerne.gif "En hvit sekskantet stjerne spretter rundt på en svart bakgrunn.")
+
++ **Vanskelig**: Trekanten i programmet er nesten likesidet, men den
+  er litt for høy, så to av sidene er rundt 12 piksler for lange. Kan
+  du endre regnestykket `y + 100` sånn at trekanten blir likesidet? Du
+  kan bruke Pytagoras' læresetning, eller sinus-funksjonen, `sin` i
+  Processing, for å finne den riktige høyden. Funksjonen `radians` kan
+  være til hjelp for å gjøre om grader til radianer hvis du vil bruke
+  sinus-funksjonen.
+
+# Trekanter {.activity}
+
+Nå skal vi se hvordan vi kan lage trekanter hvor hvert hjørne beveger
+seg for seg selv. Da trenger vi fire variabler for hvert hjørne, to
+for posisjon og to for farten, og dette for tre hjørner som blir tolv
+variabler. Vi kunne kalt dem f.eks. `x1`, `x2`, `x3` og tilsvarende
+for `y`, `xFart` og `yFart`. Isteden skal vi bruke noe som kalles en

```

Comment:
Låneord er som regel hankjønn, rett og slett fordi de fleste substantiv er hankjønn på norsk: http://www.sprakradet.no/Vi-og-vart/Publikasjoner/Spraaknytt/Arkivet/Spraknytt-2011/Spraknytt-12011/Med-andre-ord/. Men siden array ikke er et normert låneord, er det litt vanskelig å avgjøre hva som er best. For meg er dette et ord som er naturlig å bøye i hankjønn.

Det er heller ikke så lett å finne på Google hva som er vanligst, siden Arrayet er et fransk etternavn. "En array" har omtrent ti ganger så mange treff som "et array" på Google, men disse er ikke nødvendigvis representative heller (en trenger ikke å være artikkelen, men er noen ganger språkkoden for engelsk for eksempel). Jeg får svenske og danske treff på topp når jeg bruker intetkjønn, men det er en salig blanding uavhengig av dem alle tre uavhengig av hvilket kjønn jeg velger.